### PR TITLE
Updated again to accept a variables.json argument

### PR DESCRIPTION
**Pull request by markterrill**
Hi,

Previously submitted a PR (https://github.com/spark/particle-cli/pull/225) (and sent a release) that appeared to be accepted but wasn't merged. I've gone through and re-added the code and made the necessary adjustments to deal with things like args now being an array. It'd be awesome if this time it was merged to save folk time when setting up a _lot_ of Particles.

**Background:**
As the serial command currently expects any 'extra' arguments to be the serial com device, you need to append the .json argument to the end, ie: 
particle serial wifi /dev/cu.usbmodem1411 variables.json

variables.json looking like:
{
"network":"mynetworkssid",
"security":"WPA_AES",
"password":"mySecret!"
}

### CODE REVIEW

**Comment by monkbroc** [comment]
Hi Mark,

The CLI is not in the best shape to accept new features. I'm going through a refactor of the code base and I'd rather keep your use case of being able to set up devices from a script as a requirement than merge this particular implementation. How does that sound?

**Comment by markterrill** [comment]
Sure, I understand re the major version. Is there likelihood of a dot release that will break me just copying over SerialCommand file again?

 I'll be processing batches of 1000 units in about 2 months and that'll rely on automation (possibly done by the particle factory). It'll be stressful enough without scrambling to figure out why it's not working anymore, i.e. this time the new indenting made a diff near impossible and took a while to figure out it was the arguments becoming a hash that was the issue.

Thanks!

Mark

--
Mark Terrill | 0412 704 411

On 21 Jun 2016, at 07:28, Julien Vanier <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi Mark,

The CLI is not in the best shape to accept new features. I'm going through a refactor of the code base and I'd rather keep your use case of being able to set up devices from a script as a requirement than merge this particular implementation. How does that sound?

**Comment by jonlogan** [comment]
Hey @markterrill, could you ping me at jon@particle.io?   I'm director of customer success at Particle and would love to learn more about what you're trying to accomplish regarding the CLI and your mfg process.  Thanks!

**Comment by markterrill** [comment]
Hi, can we merge this PR?
From memory I contacted you kind folk and the mfg line changes couldn't be made, so its up to us to make it easier for our mfg partners to update/claim/flash.
It's a very simple PR that pulls in the WiFi creds from a json file, which makes infinite sense.

File: commands/SerialCommand.js [hunk 1]
```diff
@@ -769,16 +825,23 @@ SerialCommand.prototype = extend(BaseCommand.prototype, {
...
-				message: 'Wi-Fi Password',
-				validate: function(val) {
-					return !!val;
-				}
-			}], function(ans) {
-				cb(ans.password + '\n', startTimeout.bind(self, 15000));
-			});
+            // Skip password prompt as appropriate
+            if (password){
+                console.log('Password: ' + password);
```
Comment (m-mcgowan) [inline commands/SerialCommand.js]: not sure we should echo this sensitive info to the console

**Pull request merged**